### PR TITLE
Contents for DS Menu

### PR DESCRIPTION
**Pull request by sumitd94**
This PR Contains the following changes
1. Refactoring of code
2. Using right file naming conventions for the components 
3. Improved the menu items design
4. Added contents for the DS Menu
5. When we click on "Add new Challenge" button, the modal now shows

This is with ref to #5

### CODE REVIEW

**Comment by Harshi7016** [comment]
Header added for Header Icon and Title for Challenge Box #6 PR  here

File: src/components/challenges/active.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      <p>
+        <span className={classNames.description}>Challenge Ends</span>:
+        <span className={classNames.descValue}>{content.endDate}</span>
+      </p>
+      <p>
+        <span className={classNames.description}>Active Participants</span>:
+        <span
+          className={classNames.descValue}
+        >{`${content.participantsCnt} Members`}</span>
+      </p>
```
Comment (ankushdharkar) [inline src/components/challenges/active.js:24]: Seeing a lot of repeated code of the format:

```jsx
return (
      <p>
        <span className={classNames.description}>{desc}</span>:
        <span className={classNames.descValue}>{val}</span>
      </p>
)
```

What are your thoughts on making it a reusable function @sumitd94?

File: src/components/modal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            Cancel
+          </button>
+          <button className={classNames.btnSuccess}>Add</button>
+        </div>
+      </div>
+    </>
+  );
+};
+
+export default Modal;
```
Comment (ankushdharkar) [inline src/components/modal/index.tsx:36]: Let's move this modal in a separate PR please. Let's focus only on our main feature like we discussed

File: src/components/navbar/index.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        <a className={page == 'Mine' ? classNames.active : ''}>Mine</a>
+      </Link>{' '}
+      |
+      <Link href='/challenges'>
+        <a className={page == 'DS' ? classNames.active : ''}>DS</a>
+      </Link>{' '}
+      |
+      <Link href='/all'>
+        <a className={page == 'All' ? classNames.active : ''}>ALL</a>
+      </Link>
```
Comment (ankushdharkar) [inline src/components/navbar/index.tsx:32]: Scope for a reusable function?

File: src/components/section/active-section/index.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import classNames from './active-section.module.scss';
+import Active from '../../challenges/active';
+
+const ActiveSection = (props) => {
+  let activeContent;
+
+  if (props.screen == 'Challenges') {
```
Comment (ankushdharkar) [inline src/components/section/active-section/index.tsx:7]: Let's use a constant to hold the value 'Challenges' in a constants file

File: src/pages/challenges.tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        id: 2,
+        title: 'Sherlock and Anagrams',
+        level: 'Easy',
+        startDate: '28 Days Ago',
+        endDate: '14 Days Ago',
+        participantsCnt: '5',
+        isActive: 0,
+      },
+    ];
+
```
Comment (ankushdharkar) [inline src/pages/challenges.tsx:43]: Let's move mock data into `mocks/challenges.json`

**Review by ankushdharkar — APPROVED** [review]

File: src/pages/challenges.tsx (hunk 5 above)
Comment (sumitd94) [reply src/pages/challenges.tsx:43]: Sure, I will move this accordingly

File: src/components/navbar/index.tsx (hunk 3 above)
Comment (sumitd94) [reply src/components/navbar/index.tsx:32]: DONE

File: src/components/section/active-section/index.tsx (hunk 4 above)
Comment (ankushdharkar) [reply src/components/section/active-section/index.tsx:7]: @sumitd94 Was this looked into?

File: src/components/label/label.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import classNames from './label.module.scss';
```
Comment (ankushdharkar) [inline src/components/label/label.js:1]: It's not just the label, right? It also has a value

File: src/components/navbar/index.tsx [hunk 7]
```diff
@@ -0,0 +1,30 @@
+import classNames from './navbar.module.scss';
+import Link from 'next/link';
+import PropTypes from 'prop-types';
+
+const Navbar = ({ page }) => {
+  const navBarContent = (title, refUrl) => {
+    const navBarText = title === 'Tasks' ? 'Index' : title;
```
Comment (ankushdharkar) [inline src/components/navbar/index.tsx:7]: What is the purpose of 'Index'?

File: src/components/section/completed-section/index.tsx [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import classNames from './completed-section.module.scss';
+import Complete from '../../challenges/complete';
+
+const CompletedSection = (props) => {
+  let completeContent;
+
+  if (props.screen == 'Challenges') {
+    completeContent = props.sectionContent.map((challenge) => {
```
Comment (ankushdharkar) [inline src/components/section/completed-section/index.tsx:8]: I think `.filter` is a better candidate. The expectation from `.map` is that you'll get the same number of elements back, that you gave in the original array

File: src/components/section/completed-section/index.tsx [hunk 9]
```diff
@@ -0,0 +1,17 @@
+import classNames from './completed-section.module.scss';
+import Complete from '../../challenges/complete';
+
+const CompletedSection = (props) => {
+  let completeContent;
+
+  if (props.screen == 'Challenges') {
```
Comment (ankushdharkar) [inline src/components/section/completed-section/index.tsx:7]: Why's this component checking what screen it's in? Should it not be the screen's responsibility to render the correct component?

File: src/pages/challenges.tsx [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+  const btnClickHandler = () => {
+    const showModalVal = state.showModal;
+    setState({ showModal: !showModalVal });
+  };
+
+  const closeModal = () => {
+    setState({ showModal: false });
+  };
+
```
Comment (ankushdharkar) [inline src/pages/challenges.tsx:21]: Let's move the modal code to separate PR please @sumitd94

File: src/components/navbar/index.tsx (hunk 7 above)
Comment (sumitd94) [reply src/components/navbar/index.tsx:7]: When we click on the `Tasks` menu, it redirects us to the Index page .. So I am assuming that when we hit the url roadmap.realdevsquad.com/
the Index page will contain the Tasks details (The Index page)

File: src/components/challenges/active.js [hunk 11]
```diff
@@ -1,6 +1,7 @@
 import { useState } from 'react';
 import classNames from './styles.module.scss';
-import Label from '../label/label';
+import Details from './details/details';
```
Comment (ankushdharkar) [inline src/components/challenges/active.js:3]: Nit: Avoid pattern like this, it's redundant and adds no major value
I personally prefer: `./details` (/details/index.js)

File: src/components/section/index.tsx [hunk 12]
```diff
@@ -0,0 +1,20 @@
+import ActiveSection from './active-section';
+import CompletedSection from './completed-section';
+import classNames from './section.module.scss';
+import PropTypes from 'prop-types';
+
+const Index = ({ content, screen }) => {
+  return (
+    <div className={classNames.section}>
```
Comment (ankushdharkar) [inline src/components/section/index.tsx:8]: Nit: Use `<section>` tag

File: src/pages/challenges.tsx (hunk 10 above)
Comment (ankushdharkar) [reply src/pages/challenges.tsx:22]: @sumitd94 Are you not reading the review comments?

**Review by ankushdharkar — CHANGES_REQUESTED** [review]

File: src/components/modal/index.tsx (hunk 2 above)
Comment (sumitd94) [reply src/components/modal/index.tsx]: Done, Ref : https://github.com/Real-Dev-Squad/website-roadmap/pull/8/commits/0e9582b11de84acbb05306c5964d59429c572733

File: src/pages/challenges.tsx (hunk 10 above)
Comment (sumitd94) [reply src/pages/challenges.tsx]: Removed the code related to modal
Red : https://github.com/Real-Dev-Squad/website-roadmap/pull/8/commits/6c51e8168baa5f75e481493c92f0deb48db92c9c

File: src/components/challenges/active.js (hunk 11 above)
Comment (sumitd94) [reply src/components/challenges/active.js:3]: Ok Got it, in my next PR will fix this.

File: src/components/section/index.tsx (hunk 12 above)
Comment (sumitd94) [reply src/components/section/index.tsx:8]: Sure

File: src/components/challenges/active.js (hunk 1 above)
Comment (sumitd94) [reply src/components/challenges/active.js]: Done
Ref : https://github.com/Real-Dev-Squad/website-roadmap/pull/8/commits/0e9582b11de84acbb05306c5964d59429c572733

File: src/components/label/label.js (hunk 6 above)
Comment (sumitd94) [reply src/components/label/label.js]: Ref : https://github.com/Real-Dev-Squad/website-roadmap/pull/8/commits/08fcb4a54dff1dd083ba1c2e4cc8df90ef496daf

File: src/components/section/completed-section/index.tsx (hunk 8 above)
Comment (sumitd94) [reply src/components/section/completed-section/index.tsx]: Ref : https://github.com/Real-Dev-Squad/website-roadmap/pull/8/commits/08fcb4a54dff1dd083ba1c2e4cc8df90ef496daf

File: src/components/section/active-section/index.tsx (hunk 4 above)
Comment (sumitd94) [reply src/components/section/active-section/index.tsx]: Yes
Ref : https://github.com/Real-Dev-Squad/website-roadmap/pull/8/commits/08fcb4a54dff1dd083ba1c2e4cc8df90ef496daf

File: src/components/constants/display-sections.js [hunk 13]
```diff
@@ -0,0 +1,2 @@
+const screenConstants = [];
+screenConstants['CHALLENGES_SCREEN_NAME'] = 'Challenges';
```
Comment (ankushdharkar) [inline src/components/constants/display-sections.js:2]: What are you exporting here? How is this working in the file you imported? Please recheck @sumitd94

File: src/components/section/active-section/index.tsx [hunk 14]
```diff
@@ -0,0 +1,24 @@
+import classNames from './active-section.module.scss';
+import Active from '../../challenges/active';
+import Constants from '../../constants/display-sections.js';
```
Comment (ankushdharkar) [inline src/components/section/active-section/index.tsx:3]: I don't see an export statement in the constants file

Comment (ankushdharkar) [reply src/components/section/active-section/index.tsx:3]: Also, prefer `import { CHALLENGES_SCREEN } from '../../constants/display-sections.js';`

File: src/components/section/completed-section/index.tsx [hunk 15]
```diff
@@ -0,0 +1,24 @@
+import classNames from './completed-section.module.scss';
+import Complete from '../../challenges/complete';
+import Constants from '../../constants/display-sections.js';
```
Comment (ankushdharkar) [inline src/components/section/completed-section/index.tsx:3]: Re: above usage

File: src/pages/challenges.tsx [hunk 16]
```diff
@@ -0,0 +1,45 @@
+import { useState } from 'react';
+import Layout from '../components/Layout';
+import Section from '../components/section';
+import Navbar from '../components/navbar';
+import Title from '../components/title';
+import ChallengesJson from '../../mock/challenges.json';
+import Constants from '../components/constants/display-sections.js';
```
Comment (ankushdharkar) [inline src/pages/challenges.tsx:7]: Re:

**Review by ankushdharkar — CHANGES_REQUESTED** [review]

File: src/components/section/active-section/index.tsx (hunk 14 above)
Comment (sumitd94) [reply src/components/section/active-section/index.tsx]: Done,
Ref: https://github.com/Real-Dev-Squad/website-roadmap/pull/8/commits/c5297db1ff9e6f0180a2600c4e7adb9ecad8f7a6

**Review by ankushdharkar — APPROVED** [review]
Thanks @sumitd94
